pkg/v1/remote/transport/ping.go

```diff
@@ -83,7 +85,10 @@ func ping(reg name.Registry, t http.RoundTripper) (*pingResp, error) {
 			// Potentially retry with http.
 			continue
 		}
-		defer resp.Body.Close()
+		defer func() {
```

Comment:
Can you add a comment describing why we do this? It might be confusing, and future readers would save time not having to trace the change back to this PR.